tests/lib/bin/download-file

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env any-python
+
+from argparse import ArgumentParser
+import requests
+import os
+try:
+    from urlparse import urlsplit
+except ImportError:
+    from urllib.parse import urlsplit
+
+
+MAX_ATTEMPTS = 5
+CONNECTION_TIMEOUT = 10
+READ_TIMEOUT = 5
+
+
+def get_url_filename(url):
+    return os.path.basename(urlsplit(url)[2])
+
+
+def download_file(url, print_file=False, output_name='', output_dir=''):
```

Comment:
Silly question (maybe missed stuff because of my vacation) - curl has --retry and --retry-delay and --connect-timeout so I wonder a bit about the benefits of python here? The code is simple but doing in python means we need to think about things like how to get the data on disk. The current code seems to be reading it into memory first and then writes it on disk which may be a problem for big files. Anyway, I guess I wonder if we could get away with using curl here. Having the abstraction of bin/download-file is nice though because it means we can replace the implementation if we want to.